src/test/java/com/mfarag/todo/TasksTest.java

```diff
@@ -0,0 +1,71 @@
+package com.mfarag.todo;
+
+import org.junit.Test;
+
+import java.util.Date;
+
+import static org.junit.Assert.*;
+
+public class TasksTest {
+    private Task TEST_TASK_1 = new Task ("New Task 1") ;
+    private Task TEST_TASK_2 = new Task ( "New Task 2") ;
+    @Test
+    public void TestReturnNoTasksIfNoneExists() {
+
+        assertNull(new Tasks().getCurrent());
+    }
+
+    @Test
+    public void TestReturnTaskIfTaskExists() {
+        Tasks tasks = new Tasks() ;
+        tasks.addTask(TEST_TASK_1);
+        assertNotNull(tasks.getCurrent());
+
+
+    }
+
+    @Test
+    public void TestReturnFirstTaskIfMultipleTasksExist(){
+        Tasks tasks = new Tasks() ;
+        tasks.addTask(TEST_TASK_1);
+        tasks.addTask(TEST_TASK_2);
+
+        assertEquals(TEST_TASK_1, tasks.getCurrent());
+
+    }
+
+    @Test
+    public void TestReturnFirstUncompletedTask () {
+        Tasks tasks = new Tasks() ;
+        tasks.addTask(TEST_TASK_1);
+        tasks.addTask(TEST_TASK_2);
+
+        Task current = tasks.getCurrent() ;
+        current.complete();
+        //assertFalse(TEST_TASK_2.isCompleted());
+        assertEquals(new Date(),current.isCompleted());
+
+    }
+
+    @Test
+    public void TestCompletedContainDate () {
+        Tasks tasks = new Tasks() ;
+        tasks.addTask(TEST_TASK_1);
+        tasks.addTask(TEST_TASK_2);
+
+        Task current = tasks.getCurrent() ;
+        current.complete();
+
+        assertEquals(new Date(),current.isCompleted());
```

Comment:
Contains the correct date... the date you were expecting?

Reply:
Testing the correct date is tricky...
Maybe check this StackOverflow question: https://stackoverflow.com/questions/1671001/compare-date-objects-with-different-levels-of-precision